src/Model/Behavior/TrashBehavior.php

```diff
@@ -180,7 +180,27 @@ public function trash(EntityInterface $entity, array $options = []): bool
      */
     public function beforeFind(EventInterface $event, SelectQuery $query, ArrayObject $options, bool $primary): void
     {
+        $option = $query->getOptions();
+        if (!empty($option['skipAddTrashCondition'])) {
+            return;
+        }
+
         $field = $this->getTrashField();
+
+        if ($this->shouldAddTrashCondition($query, $field)) {
+            $query->andWhere($query->newExpr()->isNull($field));
```

Comment:
```suggestion
            $query->andWhere($field . ' IS' => null);
```
This is easier to grok.